qa/tasks/vstart_runner.py

```diff
@@ -546,14 +552,14 @@ def get_keyring_path(self):
         else:
             return keyring_path
 
-    def run_shell(self, args, wait=True, stdin=None, check_status=True,
-                  omit_sudo=False):
+    def run_shell(self, args, wait=True, stdin=None, stdout=None, stderr=None,
+                  check_status=True, omit_sudo=False):
         # FIXME maybe should add a pwd arg to teuthology.orchestra so that
         # the "cd foo && bar" shenanigans isn't needed to begin with and
         # then we wouldn't have to special case this
         return self.client_remote.run(args=args, wait=wait, cwd=self.mountpoint,
-                                      stdin=stdin, check_status=check_status,
-                                      omit_sudo=omit_sudo)
+                                      stdin=stdin, stdout=stdout, stderr=stderr,
+                                      check_status=check_status, omit_sudo=omit_sudo)
```

Comment:
Overall run_shell can be dropped, use remote.sh or remote.run correspondingly if you need output right away or not.